### ISSUE TITLE
Investigate alternatives to HTTP download

### ISSUE THREAD

**Issue by dkirkby** [dkirkby/bossdata#16]
This [internal trac page](https://trac.sdss.org/wiki/DataArchive/DataTransfer) recommends using `rsync` or `globus` for better throughput.

**Comment by dkirkby** [comment]
Public info about alternatives is also at http://www.sdss.org/dr12/data_access/bulk/

**Comment by dkirkby** [comment]
`bossfetch` now has a `--save` option that saves a list of remote file names of the form `/sas/...` to a file and then exits without downloading anything (*), so this is hopefully a good starting point for adding a `--globus` flag that will write something that globus can use directly.

(*) That's not strictly true since the small `spPlancomb` files will be downloaded, if necessary, in order to determine which `spFrame` or `spCFrame` files need to be downloaded.  This only happens with either of the `--frame` or `--cframe` options is passed to `bossfetch`.

**Comment by dmargala** [comment]
`bossfetch` would additionally need the to/form endpoints that have been registered with `globus`.

Here is an example of how this could be used:

```
bossfetch qso.dat --globus-from lbnl#sdss3 --globus-to dmargala#macbookpro --save globus_qso_transfer.dat
```

Where the contents of `globus_qso_transfer.dat` should look like:

```
lbnl#sdss3/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/4216/spec-4216-55477-0758.fits dmargala#macbookpro/Users/daniel/data/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/4216/spec-4216-55477-0758.fits
lbnl#sdss3/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/4216/spec-4216-55477-0228.fits dmargala#macbookpro/Users/daniel/data/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/4216/spec-4216-55477-0228.fits 
...
```

Then to transfer the files the user would run:

```
cat globus_qso_transfer.dat | ssh dmargala@cli.globusonline.org transfer --label 'qso file transfer'
```

Some information on how to get started with globus and it's command line interface:
- [Getting started with the Command Line Interface (CLI)](https://support.globus.org/entries/21245717)
- [Using the Command Line Interface (CLI)](https://support.globus.org/entries/30058643?)
- [CLI: Beyond the basics](https://support.globus.org/entries/29642203-CLI-Beyond-the-basics)
- [CLI Command Reference: transfer](https://support.globus.org/entries/38959537-CLI-Command-Reference-transfer)

**Comment by dkirkby** [comment]
For the command-line args, I suggest a single option for both endpoints to simplify the args error checking (and less typing):

```
bossfetch qso.dat --globus lbnl#sdss3:dmargala@macbookpro --save globus_qso_transfer.dat
```

Will that `ssh` command automatically skip files that already exist locally?  If so, does it require that the file sizes match?  What file ownership and permissions will copied files have?

**Comment by dmargala** [comment]
I setup a endpoint on darkmatter to test this (will document this process elsewhere). The following `ssh` command works and will skip files if the file sizes match.

```
ssh dmargala@cli.globusonline.org transfer -s 1 < globus_test.dat
```

The different `SYNC_LEVEL` options:

```
-s SYNC_LEVEL         Only transfer files that have changed. SYNC_LEVEL:
                        0=exist, 1=size, 2=timestamp, 3=checksum
```

Maybe we should even check the timestamp? 

(Including the `--label` option would not work for some reason...)

File ownership and permissions:

```
$ ls -lh /data/sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/4216/
total 432K
-rw-r--r--+ 1 dmargala dmargala 214K Jun 17 22:57 spec-4216-55477-0228.fits
-rw-r--r--+ 1 dmargala dmargala 214K Jun 17 22:57 spec-4216-55477-0758.fits
```

**Comment by dkirkby** [comment]
Fixed by #30
